app/models/payment/braintree_payment_method.rb

```diff
@@ -1,3 +1,4 @@
 class Payment::BraintreePaymentMethod < ActiveRecord::Base
-  belongs_to :customer, class_name: Payment::BraintreeCustomer, foreign_key: :customer_id
+  belongs_to :customer,     class_name: Payment::BraintreeCustomer
```

Comment:
So basically the change is to use the surrogate primary key as the FK instead of the customer_id?